### PR TITLE
Add Verify snippets for workflows, pin codes, and search

### PR DESCRIPTION
**Pull request by lornajane**
I've added an example to show use of search, which was not included in the existing set of snippets.

There are also two new features for Verify that I added examples for:

* Choosing a workflow - the docs on the workflows are here https://developer.nexmo.com/verify/guides/workflows-and-events
* Supplying your own PIN Code - this is a limited feature and will go into the limited docs and NOT onto NDP but I thought the code sample might as well go in the usual place in case we wanted it.

Related: I also added specs for the verify snippets as we didn't have them already (https://github.com/Nexmo/code-snippet-specs/pull/16)

### CODE REVIEW

File: verify/send-verification-code-with-workflow.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+source "../config.sh"
+
+# Try workflow_id 4 for SMS followed by SMS
```
Comment (marklewin) [inline verify/send-verification-code-with-workflow.sh]: Does that comment really help? "SMS followed by SMS" is confusing.

Comment (lornajane) [reply verify/send-verification-code-with-workflow.sh]: Hmm, it's a new field and I wanted to give an example. Workflow 4 is my favourite as it does SMS-> SMS ... can we do better with the comment wording or would you prefer to just remove it and people can use the docs to find out what the secret integers mean?

Comment (lornajane) [reply verify/send-verification-code-with-workflow.sh]: I removed the comment and switched to a config value, does this make more sense?

Comment (marklewin) [reply verify/send-verification-code-with-workflow.sh]: Yes, I think this is a much better approach!

**Pull request merged**